Source/Core/Core/HW/SI/SI.cpp

```diff
@@ -611,7 +995,21 @@ static void ChangeDeviceDeterministic(SIDevices device, int channel)
   CoreTiming::ScheduleEvent(SystemTimers::GetTicksPerSecond(), s_change_device_event, channel);
 }
 
-void UpdateDevices()
+bool SiIsChannelPollEnabled(u32 channel)
+{
+  std::array<bool, 4> enabled_bits = {
+      static_cast<bool>(s_poll.EN0),
+      static_cast<bool>(s_poll.EN1),
+      static_cast<bool>(s_poll.EN2),
+      static_cast<bool>(s_poll.EN3),
+  };
+  if (channel < 3)
```

Comment:
This will not access the last element in the array above. Intended? (If so, it could maybe use a comment as to why)

Reply:
Nice catch, thanks (should be 4). I went through and changed all the literal 4's that are used for this purpose to `MAX_SI_CHANNELS`.